koku/masu/api/upgrade_trino/util/verify_parquet_files.py

```diff
@@ -0,0 +1,313 @@
+                self.file_tracker.set_state(s3_object_key, cstates.found_s3_file)
+                local_file_path = os.path.join(self.local_path, os.path.basename(s3_object_key))
+                LOG.info(
+                    log_json(
+                        self.provider_uuid,
+                        msg="Downloading file locally",
+                        context=self.logging_context,
+                    )
+                )
+                s3_bucket.download_file(s3_object_key, local_file_path)
+                self.file_tracker.add_local_file(s3_object_key, local_file_path)
+                self.file_tracker.set_state(s3_object_key, self._coerce_parquet_data_type(local_file_path))
+                del self.logging_context[self.S3_OBJ_LOG_KEY]
+            del self.logging_context[self.S3_PREFIX_LOG_KEY]
+
+        if self.simulate:
+            self.file_tracker.generate_simulate_messages()
+            return False
+        else:
+            files_need_updated = self.file_tracker.get_files_that_need_updated()
+            for s3_obj_key, converted_local_file_path in files_need_updated.items():
+                self.logging_context[self.S3_OBJ_LOG_KEY] = s3_obj_key
+                # Overwrite s3 object with updated file data
+                with open(converted_local_file_path, "rb") as new_file:
+                    LOG.info(
+                        log_json(
+                            self.provider_uuid,
+                            msg="Uploading revised parquet file.",
+                            context=self.logging_context,
+                            local_file_path=converted_local_file_path,
+                        )
+                    )
+                    try:
+                        s3_bucket.upload_fileobj(
+                            new_file,
+                            s3_obj_key,
+                            ExtraArgs={"Metadata": {"converter_version": CONVERTER_VERSION}},
+                        )
+                        self.file_tracker.set_state(s3_obj_key, cstates.s3_complete)
+                    except ClientError as e:
+                        LOG.info(f"Failed to overwrite S3 file {s3_object_key}: {str(e)}")
+                        self.file_tracker.set_state(s3_object_key, cstates.s3_failed)
+                        continue
+        self.file_tracker.finalize_and_clean_up()
+
+    def _perform_transformation_double_to_timestamp(self, parquet_file_path, field_names):
+        """Performs a transformation to change a double to a timestamp."""
+        if not field_names:
+            return
+        LOG.info(
+            log_json(
+                self.provider_uuid,
+                msg="Transforming fields from double to timestamp.",
+                context=self.logging_context,
+                local_file_path=parquet_file_path,
+                updated_columns=field_names,
+            )
+        )
+        table = pq.read_table(parquet_file_path)
```

Comment:
Will any of these files be larger than available memory? Just wondering if we should be using `ParquetFile.iter_batch()` instead of reading the whole file at once.

Reply:
Hmmm. Good comment, so these parquet files had to go through our batching logic to be created, so I feel like we shouldn't run into any massive parquet files during this process.